smartapp-spring/src/main/java/com/smartthings/sdk/smartapp/spring/HttpVerificationService.java

```diff
@@ -0,0 +1,140 @@
+package com.smartthings.sdk.smartapp.spring;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
+import org.bouncycastle.crypto.params.RSAKeyParameters;
+import org.bouncycastle.crypto.util.PublicKeyFactory;
+import org.bouncycastle.openssl.PEMParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.Resource;
+import org.springframework.util.FileCopyUtils;
+
+import net.adamcin.httpsig.api.Authorization;
+import net.adamcin.httpsig.api.Challenge;
+import net.adamcin.httpsig.api.DefaultKeychain;
+import net.adamcin.httpsig.api.DefaultVerifier;
+import net.adamcin.httpsig.api.KeyId;
+import net.adamcin.httpsig.api.RequestContent;
+import net.adamcin.httpsig.api.Verifier;
+import net.adamcin.httpsig.ssh.jce.KeyFormat;
+import net.adamcin.httpsig.ssh.jce.SSHKey;
+import net.adamcin.httpsig.ssh.jce.UserFingerprintKeyId;
+
+
+/**
+ * All requests should have their HTTP signature verified to ensure if request is actually from SmartThings.
+ * This class has the implementation of signature authentication and verification
+ * from https://github.com/adamcin/httpsig-java.
+ * The public key that is generated when registering a smartapp; is used to verify the request.
+ */
+public class HttpVerificationService {
```

Comment:
Would be nice to add some unit tests for this

Reply:
I added a few. They're definitely not comprehensive but given this will be done a different way in the future, I've probably spent more time than I should have on it already.